c3po/job/utils/metadata.py

```diff
@@ -26,35 +28,38 @@ def insert_metadata(raw_data):
                 data = SongData(url)
                 # This is a placeholder for until we can fetch real user details
                 user = _insert_default_user(session)
-                new_link = _insert_post(url, user, raw_data, session)
+                new_link = _insert_post(url, user, data.extraAttrs, raw_data, session)
                 if new_link:
-                    new_song = _insert_song(data.track, data.extraAttrs, session)
-                    new_link.song_id = new_song.id
-                    for artist_data in data.artists:
-                        new_artist = _insert_artist(artist_data, session)
-                        _insert_artist_song(new_artist, new_song, session)
-                        _insert_song_genre(artist_data, new_song, session)
+                    if(data.track and data.artists):
+                        new_song = _insert_song(data.track, session)
+                        new_link.song_id = new_song.id
+                        for artist_data in data.artists:
+                            new_artist = _insert_artist(artist_data, session)
+                            _insert_artist_song(new_artist, new_song, session)
+                            _insert_song_genre(artist_data, new_song, session)
+            except ServerNotFoundError:
+                print("Google API unreachable!")
+                time.sleep(30)
```

Comment:
:roll_eyes:

Reply:
The API times out sometimes :confused: 
I thought putting it to sleep temporarily and checking again for the next entry should work :sweat_smile: